### PR TITLE
Omit MachineConfig and MachineConfigPool CRDs from release payload

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
The MachineConfig and MachineConfigPool CRDs are not needed in a
HyperShift cluster and they should be removed from the UI.

This change is to omit them from the release payload applied to the
guest cluster.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # [CONSOLE-2984](https://issues.redhat.com//browse/CONSOLE-2984) 

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go [hunk 1]
```diff
@@ -68,6 +68,8 @@ var (
 		"0000_50_cluster-ingress-operator_02-deployment-ibm-cloud-managed.yaml",
 		"0000_70_cluster-network-operator_02_rbac.yaml",
 		"0000_70_cluster-network-operator_03_deployment-ibm-cloud-managed.yaml",
+		"0000_80_machine-config-operator_01_machineconfig.crd.yaml",
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go:71]: oh I thought we were excluding this already, do we have a counterpart PR to exclude them properly via profile?

**Comment by ironcladlou** [comment]
Does it matter these may already be in existing clusters (e.g. IBM)?

**Review by jhadvig — APPROVED** [review]
LGTM

File: control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go (hunk 1 above)
Comment (csrwng) [reply control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go:71]: we don't yet, but I didn't want to make that change without making sure that it won't negatively affect ROKS clusters which use the same profile

**Comment by csrwng** [comment]
> Does it matter these may already be in existing clusters (e.g. IBM)?
You're right, this won't remove them from existing clusters. Let me add a manifest to do that.

**Comment by csrwng** [comment]
updated to handle the upgrade case

**Comment by csrwng** [comment]
/hold
This is failing the single test create:
```
=== CONT  TestCreateCluster
    util.go:196: 
        failed waiting for image rollout
        Unexpected error:
            <*errors.errorString | 0xc0004ca780>: {
                s: "timed out waiting for the condition",
            }
            timed out waiting for the condition
        occurred
```

**Comment by csrwng** [comment]
So the issue is that the node tuning operator fails to start because it watches MachineConfigs and MachineConfigPools by default:
```
W0422 03:53:21.874247       1 reflector.go:324] github.com/openshift/machine-config-operator/pkg/generated/informers/externalversions/factory.go:101: failed to list *v1.MachineConfigPool: the server could not find the requested resource (get machineconfigpools.machineconfiguration.openshift.io)                                                                                                                                             
E0422 03:53:21.874548       1 reflector.go:138] github.com/openshift/machine-config-operator/pkg/generated/informers/externalversions/factory.go:101: Failed to watch *v1.MachineConfigPool: failed to list *v1.MachineConfigPool: the server could not find the requested resource (get machineconfigpools.machineconfiguration.openshift.io) 
```

**Comment by relyt0925** [comment]
/hold
Cesar could we consider the implications of this as well for the SRV-IO POCs?

**Comment by relyt0925** [comment]
/unhold
Not we are allowing the manual applying for now just for POC demos with SRV-IO but these are not valid resources in hypershift

**Pull request merged**